### PR TITLE
Refactored getIword() to atomic<bool>& isColorEnabled()

### PR DESCRIPTION
**Pull request by qbradq**

### CODE REVIEW

**Comment by qbradq** [comment]
My bad. I didn't realize there were real unit tests in this thing :+1:

**Comment by qbradq** [comment]
Is there something I have to do to get the first two CI items to execute?

**Comment by agauniyal** [comment]
I'm not sure why coveralls isn't showing coverage, I'll take a look at it. Rest is fine.

File: include/rang.hpp [hunk 1]
```diff
@@ -88,7 +89,13 @@ enum class bgB {
 	gray    = 107
 };
 
-enum class control { autoColor = 0, forceColor = 1 };
+enum class control {
+    autoColor = 0,
+    forceColor = 1,
+    bufferInput = 2,
```
Comment (agauniyal) [inline include/rang.hpp]: are we using `bufferInput = 2, immediateInput = 3, rawInput = 4` yet or are these for something else?

File: include/rang.hpp [hunk 2]
```diff
@@ -261,28 +273,29 @@ void init()
...
+    ) ? rang_implementation::setColor(os, value) : os;
 }
 
 template <typename T>
 inline rang_implementation::enableControl<T> operator<<(
   std::ostream &os, T const value)
 {
 	if (value == rang::control::forceColor) {
-		os.iword(rang_implementation::getIword()) = 1;
+        rang_implementation::isColorForced() = true;
```
Comment (agauniyal) [inline include/rang.hpp]: for about 2 minutes I was thinking why compiler isn't giving any errors about this line. That was before I saw the reference as return value 👍

File: include/rang.hpp (hunk 1 above)
Comment (qbradq) [inline include/rang.hpp]: My apologies that slipped in. That is part of my input work. I am developing it in a separate pr. I will revert these enums tomorrow. That's not what they ended up being anyway.

File: include/rang.hpp [hunk 3]
```diff
@@ -268,6 +273,8 @@ void init()
 	rang_implementation::RANG_coutbuf();
 	rang_implementation::RANG_cerrbuf();
 	rang_implementation::RANG_clogbuf();
+    rang_implementation::isColorSupported() =
+        rang_implementation::supportsColor();
```
Comment (agauniyal) [inline include/rang.hpp]: do we need this? isn't `supportsColor()` method already cached as static value.

File: include/rang.hpp (hunk 1 above)
Comment (agauniyal) [inline include/rang.hpp]: sure, no worries 😄

**Comment by qbradq** [comment]
Code review comments addressed. Couldn't sleep for thinking about this :)

File: include/rang.hpp [hunk 4]
```diff
@@ -111,13 +115,19 @@ namespace rang_implementation {
 		return pLogbuff;
 	}
 
-	inline int getIword()
+#if defined(OS_LINUX) || defined(OS_MAC)
+    inline const char * getTerm() {
```
Comment (agauniyal) [inline include/rang.hpp]: I think you forgot to use this somewhere :D

Comment (qbradq) [inline include/rang.hpp]: Wow, I'm really embarrassed by this PR o_O . I refactored this down to the static variable inside supportsColor() .

**Comment by agauniyal** [comment]
squashed the commits into one to make sure checks were passing.

**Pull request merged**